### DISCUSSION TITLE
prometheus metrics on devices

### DISCUSSION THREAD

**Discussion by shiftux** [arendst/Tasmota#13010] [Q&A] [answered]
Hi,

following mostly [this guide](https://www.markhansen.co.nz/monitoring-electricity-with-tasmota/) I compiled a new version of Tasmota including a `/metrics` endpoint. This is working for my Sonoff devices running Tasmota and I can scrape them from Prometheus and show graphs in Grafana.

However this does not seem to be working on Shelly devices. Specifically I flashed my Prometheus-enabled firmware to a Shelly1 and the `/metrics` endpoint does not respond. 

Have Prometheus metrics only been implemented for specific devices in Tasmota? - It's a bit hard to find more info regarding Prometheus in here or in https://tasmota.github.io/ ... or am I missing something obvious?

Thanks and best

**Comment by barbudor** [comment]
I see no specifics in the Prometheus river that would make non working.
Did you flashed the same binary to your Shelly as on the Sonoff ? 

When you say the /metrics endpoint doesn't respond, how did you tested that ? In a browser with http://ip.of.shelly/metrics ?

**Comment by shiftux** [comment]
yes, on both devices I flashed a self-compiled binary which sets `#define USE_PROMETHEUS` in the `tasmota/user_config_override.h` file.

yes, i tested it in a browser looking at the output of http://ip.of.shelly/metrics. but i phrased it wrongly above. it does respond, but with an empty response. curl gives the output: `curl: (52) Empty reply from server` which makes it more clear.

**Reply by barbudor** [reply]
ok, I'm compiling it now and will report soon

**Reply by barbudor** [reply]
Just to be sure:
- Are you aware of recent changes to HTTP API related to SetOption128 ?

Some security has been added recently to prevent access to HTTP API when no password is set
If SetOption128 is OFF and no Referer is provided, access is rejected
You must either have a password + proper Referer header or set SetOption128 ON

May be the 1st device is running a Tasmota version before this was introduced

EDITED: You must have password + referer if SO128=OFF

**Comment by shiftux** [comment, marked as answer]
> Just to be sure:
> 
> * Are you aware of recent changes to HTTP API related to SetOption128 ?
> 
> Some security has been added recently to prevent access to HTTP API when no password is set
> If SetOption128 is OFF and no Referer is provided, access is rejected
> You must either have a password + proper Referer header or set SetOption128 ON
> 
> May be the 1st device is running a Tasmota version before this was introduced
> 
> EDITED: You must have password + referer if SO128=OFF

 
Indeed! - That did the trick! :)
thanks for your very quick help!

**Reply by barbudor** [reply]
Just to understand, was SO128 already ON on the 1st device ?
Or was it based on an older Tasmota code ?

**Reply by shiftux** [reply]
SO128 was on. but admin PW was not set.
